### PR TITLE
[feat] Make full row in the politician table clickable

### PR DESCRIPTION
**Pull request by kl-cb**
**What changed? Why?**

Allow user to click anywhere on the politician table row to navigate to politician page

**UI changes**
None

**PlanetScale Deploy Request**

**Notes to reviewers**

**How has it been tested?**

- [x] Locally
- [x] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

**Change management**
type=routine 
risk=low 
impact=sev5

### CODE REVIEW

**Review by lucasrmp-cb — COMMENTED** [review]
We already tried twice to do this, first @travisbloom-cb and then me, both times we got a weird and hard to reproduce bug where the politician page would randomly open for no apparent reason.
@travisbloom-cb what do you think? Third time is the charm?

**Comment by travisbloom-cb** [comment]
Just QAed on safari mobile, same problem :(

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]
The bug is that when you click a filter on mobile, it will jump to a persons page. I assume this happens because the browser thinks the click is for the popover and the element below. We need to investigate radix github issues to see if theres a pending issue/known fix

File: src/components/app/dtsiClientPersonDataTable/dataTable.tsx [hunk 1]
```diff
@@ -169,6 +174,9 @@ export function DataTable<TData extends Person, TValue>({
                     className="cursor-pointer"
                     data-state={row.getIsSelected() && 'selected'}
                     key={row.id}
+                    onClick={() => {
+                      openWindow(getIntlUrls(locale).politicianDetails(row.original.slug))
```
Comment (travisbloom-cb) [inline src/components/app/dtsiClientPersonDataTable/dataTable.tsx:178]: we want to use router.push for this

File: src/components/app/dtsiClientPersonDataTable/columns.tsx [hunk 2]
```diff
@@ -6,40 +6,26 @@ import { isNil } from 'lodash-es'
...
-}): ColumnDef<Person>[] => [
+export const getDTSIClientPersonDataTableColumns = (): ColumnDef<Person>[] => [
   {
     accessorKey: 'fullName',
     accessorFn: row => dtsiPersonFullName(row),
     cell: ({ row }) => (
-      <LinkBox className="flex items-center gap-3">
+      <div className="flex items-center gap-3">
         <DTSIAvatar person={row.original} size={40} />
-        <InternalLink
```
Comment (travisbloom-cb) [inline src/components/app/dtsiClientPersonDataTable/columns.tsx:36]: will it break if we leave this? better for accessibility :)

File: src/components/app/dtsiClientPersonDataTable/globalFiltersUtils.tsx [hunk 3]
```diff
@@ -108,9 +108,18 @@ export function GlobalFilters({
           <span className="mr-2 inline-block flex-shrink-0 font-bold">Stance on crypto</span>
           <SelectValue />
         </SelectTrigger>
-        <SelectContent>
+        <SelectContent
+          // Short term fix to prevent clicking on the politician table row when user selects an option from the select menu.
+          // Issue is from radix-ui https://github.com/radix-ui/primitives/issues/1658
+          ref={ref => {
```
Comment (travisbloom-cb) [inline src/components/app/dtsiClientPersonDataTable/globalFiltersUtils.tsx:114]: O_o

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]

**Comment by travisbloom-cb** [comment]
Appears to still be broken on chrome https://swc-web-git-kaerah-politician-row-click-stand-with-crypto.vercel.app/politicians :(

File: src/components/app/dtsiClientPersonDataTable/columns.tsx (hunk 2 above)
Comment (kl-cb) [reply src/components/app/dtsiClientPersonDataTable/columns.tsx]: Added it back

**Comment by kl-cb** [comment]
> Appears to still be broken on chrome https://swc-web-git-kaerah-politician-row-click-stand-with-crypto.vercel.app/politicians :(

I cannot reproduce the bug on iphone safari and chrome, and android chrome. @travisbloom-cb  Can your share what steps you are doing to get the bug?

**Comment by travisbloom-cb** [comment]
> > Appears to still be broken on chrome https://swc-web-git-kaerah-politician-row-click-stand-with-crypto.vercel.app/politicians :(
> 
> I cannot reproduce the bug on iphone safari and chrome, and android chrome. @travisbloom-cb Can your share what steps you are doing to get the bug?

Go to [this url](https://swc-web-git-kaerah-politician-row-click-stand-with-crypto.vercel.app/politicians) on chrome, set your browser dimensions to iPhone XR, scroll down so the filters are fixed over the table, and then select the stance on crypto filter and select "Pending".


https://github.com/Stand-With-Crypto/swc-web/assets/153657370/a49da53e-d754-4bd0-b6ec-d83288dba3ee

**Comment by kl-cb** [comment]
@travisbloom-cb I followed your steps and after i did a hard refresh on the page, i did not reproduce the bug. I think it does not have the latest code changes from when you last accessed the page.


https://github.com/Stand-With-Crypto/swc-web/assets/108414827/f4757c4c-8b20-49b5-a448-804514ca5956

**Comment by travisbloom-cb** [comment]
> @travisbloom-cb I followed your steps and after i did a hard refresh on the page, i did not reproduce the bug. I think it does not have the latest code changes from when you last accessed the page.
> 
>  Screen.Recording.2024-03-13.at.12.00.52.PM.mov

Super weird. Lets sync on it during our one on one.

**Review by travisbloom-cb — APPROVED** [review]

**Pull request merged**